src/vmspawn/vmspawn.c

```diff
@@ -601,15 +615,86 @@ static int find_initrd(char **ret_initrd) {
         return -ENOENT;
 }
 
+static int start_virtiofsd(sd_bus *bus, const char *scope, const char *directory, char **ret_state_tempdir, char **ret_sock_name) {
+        _cleanup_(rm_rf_physical_and_freep) char *state = NULL;
+        _cleanup_strv_free_ char **cmdline = NULL, **cleanup = NULL;
+        _cleanup_free_ char *virtiofsd = NULL, *sock_path = NULL, *sock_name = NULL, *rm_path = NULL, *unit_name = NULL;
+        int r;
+
+        assert(bus);
+        assert(scope);
+        assert(directory);
+        assert(ret_state_tempdir);
+        assert(ret_sock_name);
+
+        r = find_executable("virtiofsd", &virtiofsd);
+        if (r < 0 && r != -ENOENT)
+                return log_error_errno(r, "Error while searching for virtiofsd: %m");
+
+        if (!virtiofsd) {
+                FOREACH_STRING(dir, "/usr/libexec/virtiofsd", "/usr/libexec/virtiofsd") {
+                        if (access(dir, X_OK) >= 0) {
```

Comment:
is this a file or a dir? you named the variable "dir", but afaics this actually the binary file?

Reply:
Yep these are files, I will change the name.

Reply:
Fixed in eae47ee9d9.